### PR TITLE
Collate only if `--validator` is given

### LINKED ISSUE #152: Only collate when `--validator` is given as CLI argument
We should only collate when the user passed `--validator` to the node as CLI argument.

### PR DESCRIPTION
**Pull request by cecton**
Fixes #152

### CODE REVIEW

File: .rustfmt.toml [hunk 1]
```diff
@@ -1 +1,2 @@
 hard_tabs=true
+merge_imports=true
```
Comment (cecton) [inline .rustfmt.toml:2]: Ah so rustfmt does have something to merge imports? cool!

(I didn't add that line. Well.. I can't remember I did it)

Comment (bkchr) [reply .rustfmt.toml:2]: I added this in the rococo branch.

Why does this contains rococo changes?

Comment (cecton) [reply .rustfmt.toml:2]: Because we also need to get rococo changes on cumulus branch right?

**Comment by rphmeier** [comment]
Nice, so this enables regular full nodes?

**Comment by cecton** [comment]
Yes!! ^_^ if I can make the ci pass...

**Comment by cecton** [comment]
Actually it also does light node if asked to do so. I will remove that because I think Basti asked me to do only full node for now

**Comment by cecton** [comment]
Right!! Thanks

File: test/parachain/src/service.rs [hunk 2]
```diff
@@ -87,84 +105,126 @@ pub fn run_collator(
...
-		client,
-		announce_block,
-		block_announce_validator,
-	);
-
-	polkadot_config.informant_output_format = OutputFormat {
-		enable_color: true,
-		prefix: format!("[{}] ", Color::Blue.bold().paint("Relaychain")),
+	let finality_proof_provider =
+		GrandpaFinalityProofProvider::new_for_service(backend.clone(), client.clone());
```
Comment (bkchr) [inline test/parachain/src/service.rs:130]: This can be removed

File: test/parachain/src/service.rs [hunk 3]
```diff
@@ -87,84 +105,126 @@ pub fn run_collator(
...
 
-	Ok(service)
+	let _rpc_handlers = sc_service::spawn_tasks(sc_service::SpawnTasksParams {
+			on_demand: None,
+			remote_blockchain: None,
+			rpc_extensions_builder: Box::new(|_| ()),
+			client: client.clone(),
+			transaction_pool: transaction_pool.clone(),
+			task_manager: &mut task_manager,
+			telemetry_connection_sinks: sc_service::TelemetryConnectionSinks::default(),
```
Comment (bkchr) [inline test/parachain/src/service.rs:160]:
```suggestion
			telemetry_connection_sinks: Default::default(),
```

Comment (cecton) [reply test/parachain/src/service.rs]: done

File: test/parachain/src/service.rs (hunk 2 above)
Comment (cecton) [reply test/parachain/src/service.rs]: done

**Pull request merged**